content/documentation/pages/2-concepts/1-architecture.md

```diff
@@ -1,33 +1,330 @@
+Data Flow sets the appropriate properties when deploying the stream so that the `source` can communicate with the `sink` over the messaging middleware.
+
+#### Streams with multiple inputs and outputs
+
+Sources, Sink, and Processors all have a single output, input or both.
+This is what makes it possible for Data Flow to set application properties that pair together one output destination to another input destination.
+However, a message processing application could have more than one input or output destination.
+Spring Cloud Stream supports this by allowing the user to define a custom binding interface.
+
+To define a stream that contains an application with multiple inputs and/or you must register the application using the type `app` instead of the type `source`, `sink` or `processor`.
+The stream definition uses the **Stream Application DSL**, which replaces the single pipe symbol `|` with the double pipe symbol `||`.
+Think of `||` as meaning 'in parallel', with no implied connection between the applications:
+
+A fictional 'orderStream' is shown below.
+
+```
+dataflow:> stream create --definition "orderGeneratorApp || baristaApp || hotDrinkDeliveryApp || coldDrinkDeliveryApp" --name orderStream
+```
+
+When defining a Stream using the `|` symbol, Data Flow can configure each application in the stream to communicate with its neighboring application in the DSL since there is always one output paired to one input.
+When using the `||` symbol, you must provide configuration properties that pair together the multiple output and input destinations.
+
+<!--NOTE-->
+
+It is also possible to create a stream with a single application using the Stream Application DSL as well as deploy an application that does not use messaging middleware.
+
+<!--END_NOTE-->
+
+These examples give you a general sense of the long lived application types.
+Additional guides will go into more detail on how to develop, test and register long-lived applications and how to deploy them.
+
+In the next major section, we will discuss the runtime architecture of the deployed Stream.
+
+### Short-lived applications
+
+Short-lived applications execute for a period of time, often minutes to hours, and then terminate.  
+Their execution maybe based on a schedule, for example execute at 6pm every weekday, or in response to an event, for example a file being put in an FTP server.
+
+The Spring Cloud Task framework allows a user to develop a short lived microservice that records the life cycle events of a short lived application such as the start time, end time and the exit code.
+
+A task application is registered with Data Flow using the name `task` to describe the type of application.
+
+For example, the shell syntax for registration of a `timestamp` task (an application that prints the current time and exits) is
+
+```
+dataflow:> app register --name timestamp --type task --uri maven://org.springframework.cloud.task.app:timestamp-task:1.3.0.RELEASE
```

Comment:
Can we use this without `dataflow:>` so that the actual command can be copied as it is?

Reply:
I was thinking that here, in the start, it made sense to show it as a shell command (they aren't going to type it in, it isn't a getting started guide), as it maybe the first time someone sees the shell.  
Long term I'd like to get some styling along the lines of this - https://docs.microsoft.com/en-us/azure/dev-spaces/quickstart-java#prepare-the-application

the `cmd` in the upper left top border of the code snippit tells you the type of content, we could have `Data Flow Shell` there.